### PR TITLE
chore: enhance error handling in GitHub Actions workflows

### LINKED ISSUE #191: Improve error handling for commit/PR creation failures in GitHub Actions
### What
Our GitHub Action is throwing generic errors (that come from e.g. `gh` cli) when it fails to create a commit or pull request. We need to enhance the error message to provide more context and actionable information.

### Why
- Better error messages help users quickly identify and resolve issues
- Reduces support requests and debugging time
- Improves overall user experience with our action

### How
1. Identify specific failure scenarios (e.g., permissions, rate limits, conflicts)
2. Catch these scenarios in our code and provide custom error messages
3. Include potential next steps or troubleshooting tips in the error output
4. Consider adding links to relevant documentation

Example improved error:
"Failed to create PR: Insufficient permissions. Ensure the GitHub token has 'repo' scope. See: [link to docs]" - or smth like that.

### PR DESCRIPTION
**Pull request by partik03**
may close #191 

- feat: Added custom error messages for permission and dependency issues.
- feat: Improved guidance for semantic PR title validation in Lint PR.
- feat: Enhanced error handling for build, test, and release steps with actionable feedback.
- chore: Included instructions on resolving common failure scenarios.

### CODE REVIEW

**Comment by partik03** [comment]
@maxprilutskiy kindly review this PR

File: .github/workflows/pr-check.yml [hunk 1]
```diff
@@ -22,13 +22,13 @@ jobs:
...
           ref: ${{github.event.pull_request.head.sha}}
           fetch-depth: 0
-          
+
       - name: Check for [skip i18n]
         run: |
           COMMIT_MESSAGE=$(git log -1 --pretty=%B)
           if echo "$COMMIT_MESSAGE" | grep -iq '\[skip i18n\]'; then
             echo "Skipping i18n checks due to [skip i18n] in commit message."
-            exit 1
```
Comment (maxprilutskiy) [inline .github/workflows/pr-check.yml:31]: damn, this is a good catch, well done!

File: .github/workflows/pr-check.yml [hunk 2]
```diff
@@ -54,7 +54,11 @@ jobs:
             ${{ runner.os }}-pnpm-store-
 
       - name: Install deps
-        run: pnpm install
+        run: |
+          if ! pnpm install; then
+            echo "Failed to install dependencies. Check your pnpm configuration or network connectivity." >&2
+            exit 1
+          fi
```
Comment (maxprilutskiy) [inline .github/workflows/pr-check.yml]: additional check is not needed here

File: .github/workflows/pr-check.yml [hunk 3]
```diff
@@ -64,11 +68,23 @@ jobs:
...
 
       - name: Require changeset to be present in PR
         if: github.event.pull_request.user.login != 'dependabot[bot]'
-        run: pnpm changeset status --since origin/main
+        run: |
+          if ! pnpm changeset status --since origin/main; then
+            echo "Error: No changeset found for this PR. Please add a changeset before merging. See: [link to changeset documentation]" >&2
+            exit 1
+          fi
```
Comment (maxprilutskiy) [inline .github/workflows/pr-check.yml]: additional checks is not needed here

File: .github/workflows/pr-lint.yml [hunk 4]
```diff
@@ -15,6 +15,14 @@ jobs:
...
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        continue-on-error: true
+      - name: Check for Errors
+        if: failure()
+        run: |
+          echo "Semantic PR title validation failed. Please ensure your PR title follows the required format, such as 'feat: add new feature' or 'fix: correct a bug'."
+          echo "Refer to the documentation on semantic PR titles: [https://www.conventionalcommits.org/en/v1.0.0/]"
+          exit 1
```
Comment (maxprilutskiy) [inline .github/workflows/pr-lint.yml]: additional check is not needed here

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -15,12 +15,13 @@ jobs:
...
       - name: Checkout
         uses: actions/checkout@v4
+
       - name: Check for [skip i18n]
         run: |
           COMMIT_MESSAGE=$(git log -1 --pretty=%B)
           if echo "$COMMIT_MESSAGE" | grep -iq '\[skip i18n\]'; then
             echo "Skipping i18n checks due to [skip i18n] in commit message."
-            exit 1
+            exit 0
```
Comment (maxprilutskiy) [inline .github/workflows/release.yml:24]: that's a good one

**Review by maxprilutskiy — CHANGES_REQUESTED** [review]
#191 was actually addressed by https://github.com/replexica/replexica/pull/274, for the most part, so allthese changes are not needed anymore

good job updating `exit 1` -> `exit 0`, but the rest is not needed, as essentially it just more code that wouldn't change developer experience much

1. leave exit 1 -> exit 0 changes
2. leave the rest as it was
3. let's merge it

File: .github/workflows/pr-check.yml (hunk 3 above)
Comment (partik03) [reply .github/workflows/pr-check.yml]: removed

File: .github/workflows/pr-lint.yml (hunk 4 above)
Comment (partik03) [reply .github/workflows/pr-lint.yml]: okay

File: .github/workflows/pr-check.yml (hunk 2 above)
Comment (partik03) [reply .github/workflows/pr-check.yml]: okay

**Comment by partik03** [comment]
@maxprilutskiy kindly check again

**Comment by maxprilutskiy** [comment]
@partik03 merged!

**Pull request merged**